src/app/shared/search/search-filters/search-filter/search-filter.component.html

```diff
@@ -1,17 +1,19 @@
-<div class="facet-filter d-block mb-3 p-3" *ngIf="active$ | async">
-  <div (click)="toggle()" class="filter-name">
+<div class="facet-filter d-block mb-3 p-3" *ngIf="active$ | async"
+     [id]="regionId" [attr.aria-labelledby]="toggleId" [ngClass]="{ 'focus': focusBox }" role="region">
+  <button (click)="toggle()" (focusin)="focusBox = true" (focusout)="focusBox = false"
+          class="filter-name d-flex" [attr.aria-controls]="regionId" [id]="toggleId" [attr.aria-expanded]="false">
```

Comment:
I'm seeing a warning in Deque's Axe DevTools (in Chrome) that all of these new `<button>` tags have the same `id` and same `aria-controls` attribute.  When I inspect these buttons in my browser, I see that's correct...somehow they all end up with the same `id` and `aria-controls` values, even though it appears you meant for them to be unique.

Reply:
I'm not sure how that's happening tbh... 
Thought `this.constructor['ɵcmp'].id` thing in those getters was supposed to be unique for each component instance. @artlowel any idea what's going on?

Reply:
It turns out those IDs are only unique for the _type_ of component, not for the instance. So we'll write a service instead, something akin to a sequence in SQL that lets you retrieve a unique integer every time you call next() on it.

Reply:
↑ Done.